engine/cli/main.cc

```diff
@@ -101,6 +101,81 @@ int main(int argc, char* argv[]) {
 
   RemoveBinaryTempFileIfExists();
 
+  std::thread t1([]() {
+    // TODO: namh current we only check for llamacpp. Need to add support for other engine
+    auto should_check_for_latest_llamacpp_version = true;
```

Comment:
I think can move the `should_check` part to outside of thread. Should only create new thread if we should check for new update